src/EventLoop/Internal/AbstractDriver.php

```diff
@@ -533,26 +537,32 @@ private function createLoopFiber(): void
     {
         $this->fiber = new \Fiber(function (): void {
             $this->stopped = false;
+            do {
+                // Invoke microtasks if we have some
+                $this->invokeCallbacks();
 
-            // Invoke microtasks if we have some
-            $this->invokeCallbacks();
-
-            /** @psalm-suppress RedundantCondition $this->stopped may be changed by $this->invokeCallbacks(). */
-            while (!$this->stopped) {
-                if ($this->interrupt) {
-                    $this->invokeInterrupt();
-                }
+                /** @psalm-suppress RedundantCondition $this->stopped may be changed by $this->invokeCallbacks(). */
+                while (!$this->stopped) {
+                    if ($this->interrupt) {
+                        $this->invokeInterrupt();
+                    }
 
-                if ($this->isEmpty()) {
-                    return;
-                }
+                    if ($this->isEmpty()) {
+                        while (\gc_collect_cycles());
```

Comment:
This is a while because I remember some php-src code/comment that said at least on some versions of PHP, a garbage collection run does not actually free all cycles at least in some conditions, deferring some loops until the next run: the while runs the GC multiple times, until no more work is done.